ntrans.py

```diff
@@ -1,107 +1,75 @@
 
-        data_writer.writerow(('English', 'Swedish'))
+        data_writer.writerow(("English", "Swedish"))
         for source_target_pair in source_target_pairs:
             data_writer.writerow(source_target_pair)
 
     return print("N-Trans CSV Glossary has been successfully saved to " + full_path)
 
 
-def machine_translate_ngrams(list_of_ngrams, language=None):
+def machine_translate_ngrams(ngrams):
+    """
+    Translates each N-gram and appends the source/target pair to a list.
+    """
+    source_target_pairs = []
 
     translator = translatepy.Translator()
 
-    source_target_pairs = []
-
-    for enum_count, source_string in enumerate(list_of_ngrams, start=1):
-        print(f"Translating item {enum_count}/{len(list_of_ngrams)}")
-        translated_string = translator.translate(
-            source_string, "Swedish"
-        )  # TODO: Change to language variable
-        # print(source_string + " --> " + str(translated_string))
-        source_target_pairs.append((source_string, str(translated_string).lower()))
+    for n in ngrams.keys():
+        for enum, source_ngram in enumerate(ngrams[n], start=1):
+            print(f"Translating {n}-gram no. {enum} / {len(ngrams[n])}")
+            target_ngram = str(
+                translator.translate(source_ngram, "Swedish")
+            ).lower()  # TODO: Target language should be variable-based
+            source_target_pairs.append((source_ngram, target_ngram))
 
     create_csv_file(source_target_pairs)
 
 
-def count_ngram_frequency(ngram_strings, desired_data_size):
-    x_most_common_ngrams = collections.Counter(ngram_strings).most_common(desired_data_size)
-    return [ngram[0] for ngram in x_most_common_ngrams]
-
-
-def create_ngrams(corpus_data, ngram_size, desired_data_size):
-    """
-    Strips string from special characters, creates tuple n_grams, deletes tuples
-    with numbers, joins tuples into strings
+def read_ngram_files(user_desired_ngrams=[2, 3, 4, 5, 6], data_size=500):
     """
+    Reads N-gram files depending on which N-grams the user wants to output.
 
-    # Removes punctuation and special characters from string.
-    text_without_punctuation = re.sub(r"[^\w']+", " ", corpus_data).lower()
-
-    # TODO: Make the .replaces more effective
-    text_without_punctuation = text_without_punctuation.replace(" '", "'")
-    text_without_punctuation = text_without_punctuation.replace(" n't", "n't")
+    Appends the N-grams to a dict with a size of data_size per N-gram (specified by user).
+    """
 
-    # TODO: Put ngram function in for loop to include all N-grams that the user wants.
-    # TODO: Switch place of ngram creation and joining ngram_tuples into strings, since joining ngram_tuples takes a long time.
-    # Splits string into N-grams and adds them to tuple_list.
-    generated_ngram_tuples = nltk.ngrams(text_without_punctuation.split(), ngram_size)
+    ngrams = {2: [], 3: [], 4: [], 5: [], 6: []}
```

Comment:
This code does not use `user_desired_ngrams`, so if `user_desired_ngrams` is set to `[2, 3, 4, 5, 6, 7]`, it will try to find `7` from this dict and error.

```suggestion
    ngrams = {n: [] for n in user_desired_ngrams}
```